Sources/Segment/Types.swift

```diff
@@ -277,14 +277,13 @@ extension RawEvent {
     internal func applyRawEventData(store: Store) -> Self {
         var result: Self = self
         
-        guard let system: System = store.currentState() else { return self }
         guard let userInfo: UserInfo = store.currentState() else { return self }
         
         result.anonymousId = userInfo.anonymousId
         result.userId = userInfo.userId
         result.messageId = UUID().uuidString
         result.timestamp = Date().iso8601()
-        result.integrations = system.integrations
+        result.integrations = try? JSON([String: Any]())
```

Comment:
Don't you need to leave the `self.integrations` here? Else it strips all user defined integrations before the timeline processes it

Reply:
Yeah; this happens right when the event comes in.  The user hasn't had a chance to do anything yet so there's nothing to override.